x-pack/plugins/synthetics/common/runtime_types/monitor_management/monitor_types_project.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as t from 'io-ts';
+import { ScreenshotOptionCodec } from './monitor_configs';
+
+export const ProjectMonitorThrottlingConfigCodec = t.interface({
+  download: t.number,
+  upload: t.number,
+  latency: t.number,
+});
+
+export const ProjectBrowserMonitorCodec = t.intersection([
+  t.interface({ id: t.string, name: t.string, schedule: t.number, content: t.string }),
+  t.partial({
+    locations: t.array(t.string),
```

Comment:
what happens when user doesn't provide these?

Reply:
Adjusted the logic to make it required. Now if the user doesn't pass it, it'll end up in the `failedMonitors` list.